src/sim.jl

```diff
@@ -0,0 +1,264 @@
+#!/usr/bin/env julia
+using Distributions
+using JuliaDB
+using Unitful
+
+include("RoutingKit.jl")
+
+
+
+"""
+For a given trip_id, determine how much time is spent waiting at stops during 
+that trip.
+
+Args:
+  stop_prob  - Probability of stopping at a given stop with zero duration
+  zstop_time - Seconds spent at a stop that is otherwise given zero duration
+"""
+function GetStopChargingTime(stops, trip_id; stop_prob=0.3, zstop_time=10)
+    #Get list of stops used by this trip and merge in the stop data
+    trip_stops = filter(row->row[:trip_id]==trip_id, stops)
+    trip_stops = filter(row->row[:inductive_charging]==true, trip_stops)
+    #Total time we are scheduled to be stopped
+    stopped_time = sum(trip_stops.stop_duration)
+
+    #Some of the stops with zero duration we do end up stopping at briefly
+    zero_count = sum(trip_stops.stop_duration.==0)     #Number of zero-duration stops
```

Comment:
I'm unfamiliar with Julia notation, but is that period at `stop_duration.` extraneous?

Reply:
No. That distributes the equality across the vector.